### ISSUE TITLE
Different go_package-s in google/rpc/*.proto

### ISSUE THREAD

**Issue by vlkv** [googleapis/api-common-protos#113]
Hi, I cannot understand how to correctly use google/rpc/*.proto definitions in my project. Please explain the best practices here.

I copied google/rpc/*.proto into my repository with protos. Compiled them with buf tool and got all three go modules in one directory:
`google/rpc/code.pb.go  google/rpc/error_details.pb.go  google/rpc/status.pb.go`. Each module declares different package name:
`package code`, `package errdetails`, `package status`. This results in a compilaion errror, because Go requires that all modules in one same dir must declare the same one package, not three different ones.

How am I supposed to use those google/rpc/*.proto in my project, please give an advice.

**Comment by noahdietz** [comment]
tl;dr: For Go code, use the types in https://pkg.go.dev/google.golang.org/genproto/googleapis/rpc. Don't use this repo for protos, use https://github.com/googleapis/googleapis instead.

> I copied google/rpc/*.proto into my repository with protos. 

You shouldn't copy the protos into your own project, but use these from this repository e.g. clone the repo with the protos somewhere, or use it as a git submodule. If your protos depend on google/rpc/*.proto, then when running `protoc` or another protobuf compiler, you just need to `-I` include the path to this project to make the protos resolvable. Really you should use the protos in github.com/googleapis/googleapis too.

> Compiled them with buf tool and got all three go modules in one directory:

I'm not sure what tool you are using (i've not used the "buf" tool), but `protoc --go_out=output/directory/here google/rpc/*.proto` would generated the Go code relative to that directory using the full importpath specifed in the `go_package` option e.g. `output/directory/here/google.golang.org/genproto/googleapis/google/rpc/status/status.pb.go`. I just tested this myself with `protoc`.

**Comment by vlkv** [comment]
Thanks a lot!

BTW: "buf" is this tool https://docs.buf.build/tour/generate-go-code. In the end, I could reach the same behaviour with their option `opt: paths=source_relative`.
